packages/core/src/awsService/ec2/sshKeyPair.ts

```diff
@@ -26,14 +40,38 @@ export class SshKeyPair {
         if (result.exitCode !== 0) {
             throw new ToolkitError('ec2: Failed to generate ssh key', { details: { stdout: result.stdout } })
         }
+        chmodSync(keyPath, 0o600)
     }
 
     public getPublicKeyPath(): string {
         return this.publicKeyPath
     }
 
+    public getPrivateKeyPath(): string {
+        return this.keyPath
+    }
+
     public async getPublicKey(): Promise<string> {
-        const contents = await fs.readFile(this.publicKeyPath, 'utf-8')
+        const contents = await fs.readFileAsString(this.publicKeyPath)
         return contents
     }
+
+    public async delete(): Promise<void> {
+        await fs.delete(this.keyPath)
+        await fs.delete(this.publicKeyPath)
+
+        if (!this.lifeTimeout.completed) {
+            this.lifeTimeout.cancel()
+        }
+
+        this.deleted = true
+    }
+
+    public isDeleted(): boolean {
+        return this.deleted
```

Comment:
can this extra state be avoided? usually better to check "reality", i.e. check the filesystem. after all, we're already storing the filepath.

Reply:
So in that case, replace the `isDeleted()` method with one that checks the actual filesystem?